### PR TITLE
[REF] Rename T1c to "minimum image regression"

### LINKED ISSUE #605: Rename T1c to "minimum image regression"
### Summary
In a manuscript that I'm working on, I describe, in detail, the method implemented in [`tedana.gscontrol.gscontrol_mmix()`](https://github.com/ME-ICA/tedana/blob/150b5c0dd4e452ed00861e7e8d4347f1583599c5/tedana/gscontrol.py#L111).
We generally refer to this method as either `gscontrol_mmix` (based on the array it works on), `T1c` (because it references a T1-like effect that it removes as "T1-correction"), or `MIR` (aka minimum image regression).  I don't recall the origin of the term "minimum image regression", beyond that regressing out a minimum image from a time series is very much a step in the function. I think it might have come from the original MEICA code, but please correct me if I'm wrong on that.

In any case, the method, while recommended in MEICA, is not actually documented in a publication AFAIK. This means that, unfortunately, the manuscript I'm working on might end up being the first detailed description of the method. To make things as clear as possible, I'd like to standardize how we talk about it, both in the documentation and in the code itself.

### Next Steps
1. Rename `gscontrol_mmix` to `gscontrol_mir`.
1. Incorporate detailed steps from my manuscript (still in-prep, sorry) into the docstring to make the procedure clearer.
1. Rename outputs to include `MIR` or `mir` instead of `T1c` (except for the map of the T1-like effect, of course).

### PR DESCRIPTION
**Pull request by tsalo**
Closes #605.

Changes proposed in this pull request:

- Rename function name, variables, and output files for T1c global signal regression to match new designation as "minimum image regression."
- Change multiline string-based comments to regular comments.
- Clean up some steps (e.g., z-scoring of arrays) in the function.

### CODE REVIEW

File: tedana/gscontrol.py
Comment (jbteves) [inline tedana/gscontrol.py:158]: If we're naming everything based on stat names, shouldn'lt we make this `optcom_mean` instead of `optcom_mu`? I figure since we have other style changes in this PR, this would be a good one to change this for.

File: tedana/gscontrol.py
Comment (jbteves) [inline tedana/gscontrol.py:167]: It's a little counter-intuitive to me to change this from "bold_ts" to "mehk_ts" (mid-elbow, high-kappa?) but leaving the comment the same. I'd recommend either changing the comment or reverting the variable name change.

**Review by jbteves — CHANGES_REQUESTED** [review]
Only one slight change requested, otherwise looks good!

File: tedana/gscontrol.py
Comment (tsalo) [reply tedana/gscontrol.py:167]: You're totally right. I will improve the comment. I switched to `MEHK` (multi-echo high kappa) because "BOLD" is so unclear when we're dealing with multi-echo data.

File: tedana/gscontrol.py
Comment (emdupre) [inline tedana/gscontrol.py:14]: Was this just for style ?

File: tedana/gscontrol.py
Comment (emdupre) [inline tedana/gscontrol.py:163]: What is `comp_pes` short for ?

File: tedana/gscontrol.py
Comment (emdupre) [reply tedana/gscontrol.py:167]: Thanks, I think that helps !! On the same point : can we write out MEHK here ?

File: tedana/gscontrol.py
Comment (emdupre) [inline tedana/gscontrol.py:168]: I know that this is a bigger question, but if we can maybe mention in the docstring that this is how we derive the T1-like effect ? Or should we change this comment / variable to reference the "minimum image" ?

File: tedana/gscontrol.py
Comment (emdupre) [inline tedana/gscontrol.py:173]: I'm not sure why the diff is splotting like this, but just to confirm: None of the functionality is changed here, no ?

Comment (emdupre) [reply tedana/gscontrol.py:173]: (From my look-through I'd think not, but wanted to double-check !

**Review by emdupre — COMMENTED** [review]
Thanks, @tsalo ! I think this looks great.
I had a few small clarifications / questions.

File: tedana/gscontrol.py
Comment (tsalo) [reply tedana/gscontrol.py:14]: Yeah, I ran `black` and `isort` on the file and then only kept changes from the top of the file and the `minimum_image_regression` function.

Comment (tsalo) [reply tedana/gscontrol.py:14]: Scratch that- I don't think I ran `isort`. But I _did_ run `black`.

File: tedana/gscontrol.py
Comment (tsalo) [reply tedana/gscontrol.py:163]: component parameter estimates

File: tedana/gscontrol.py
Comment (tsalo) [reply tedana/gscontrol.py:173]: Yeah, that's weird... You're correct, though, that the functionality shouldn't change.

File: tedana/gscontrol.py
Comment (jbteves) [reply tedana/gscontrol.py:168]: I think adding to the docstring would be better, because it'll also show up in the API docs.

File: tedana/gscontrol.py
Comment (jbteves) [reply tedana/gscontrol.py:173]: It's splitting like that because of the change from """ -> # 
They're all so close together that the heuristic is counting it as big text blobs since it's multi-line.

**Review by jbteves — DISMISSED** [review]
This looks good, thanks @tsalo !

File: tedana/gscontrol.py
Comment (tsalo) [reply tedana/gscontrol.py:199]: I've improved the docstring.

**Comment by tsalo** [comment]
@jbteves I hate to discard your review, but I improved the docstring and added a reference to make things clearer. I'd love to know what you and @emdupre think of the new section in Notes.

**Comment by jbteves** [comment]
I think it's a good addition, thanks!

**Review by jbteves — APPROVED** [review]

File: tedana/gscontrol.py
Comment (emdupre) [reply tedana/gscontrol.py:198]:
```suggestion
    comp_pes = np.linalg.lstsq(mmix, optcom_z.T, rcond=None)[0].T  # component parameter estimates
```

**Review by emdupre — APPROVED** [review]
This looks great !! I had one small suggestion, but it's not worth holding up this PR over. Happy to see this in :+1:

**Comment by tsalo** [comment]
Thanks @emdupre and @jbteves. I'll wait until the CI passes and then I'll merge. No need to re-review.

**Pull request merged**
